xml/snmp_sysdescr.xml

```diff
@@ -3397,7 +3397,21 @@ Copyright (c) 1995-2005 by Cisco Systems
   <!--======================================================================
                               Juniper
    =======================================================================-->
-  <fingerprint pattern="^Juniper (\S+) : (\d+\S+)$">
+  <fingerprint pattern="^NetScreen-[\w]+ version ([\w.]+) \(SN: [\d]+, Firewall\+VPN\)$">
```

Comment:
IMO, you should capture the model and assert it as `hw.model`.  This is done for one other thing in recog, and it just so happens to be related -- Juniper fingerprints via SNMP `sysDesc`